.github/workflows/lint_and_test.yml

```diff
@@ -0,0 +1,51 @@
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+name: Lint and Test
+
+defaults:
+  run:
+    shell: bash
+
+
+jobs:
+  conventional-commits:
```

Comment:
The code looks good, but I think maintaining only one ci.yaml will be easier and tidier vs trying to unedrstand what runs where. You can use the `on` keyword for individual steps as well + combintation with `condition`. 

I am basing this assumption on examples seen in the community
- Go-Chi - https://github.com/go-chi/chi/blob/master/.github/workflows/ci.yml
- Go Gin - https://github.com/gin-gonic/gin/tree/master/.github/workflows
- Swaggo https://github.com/swaggo/swag/tree/master/.github/workflows


Hence I think we should opt to use the industry standard first, adopt later only if we see there is a need.

Reply:
In the above comment, Gin and Swaggo are using multiple files, Chi is using one but not for a release.
This PR is introducing one workflow for both lint and test, so it meets your ask.